src/suggestionlistmodel.cpp

```diff
@@ -55,6 +57,7 @@ void SuggestionListModel::append(const QList<SuggestionData>& suggestionList)
 
     for (const auto& suggestion : suggestionList)
         m_suggestions.append(suggestion);
+    setNoMoreSuggestion(suggestionList.size());
```

Comment:
This is a bad way for establishing exhaustion of suggestions as it means that `SuggestionListModel::append()` must be called only on the output of `SuggestionListWorker`. A better solution (though still a hack) may be passing the information on availability of additional suggestions via a special entry (e.g. with empty url) in `suggestionList`. But a simpler solution is to maintain the information on availability of more suggestions in `SearchLineEdit` since it is the class that arranges fetching of suggestions.

Reply:
Done